frontend-react/src/pages/SignupPage.tsx

```diff
@@ -61,20 +61,42 @@ const RegisterPage = () => {
           uid: firebaseUser.uid
         }
       })
-      // TODO: 後続タスクで処理を追加する
+
+      setUser(firebaseUser)
+      // TODO: 後続タスクでhome画面を追加する際修正
       console.log("home画面は次のissueで作成予定！")
     } catch (error: unknown) {
       if (error instanceof FirebaseError) {
         setError(getErrorMessage(error))
+        setUser(null)
         return
+      }  
+
+      const responseError = error as RailsApiError
+      if (responseError.response?.status === 422) {
+        const errorData = responseError.response?.data
+
+        if (errorData?.errors?.email?.includes("このメールアドレスは既に存在します。")) {
+          setUser(firebaseUser ?? null)
```

Comment:
以下だとダメなケースってどんな時がありますかね？？
railsへapiをリクエストしている時点でfirebaseは成功しているということになるのでnullになるケースがどういう時か気になり👀

もし、nullになるケースがないのであればnull考慮は外したほうが良いかなと思います
理由としては他のエンジニアがこの実装を始めて見た時にnullになるケースはどういう時かを調べないといけなくなるため混乱の元だからです。
```suggestion
          setUser(firebaseUser)
```